### PR TITLE
HW_8: oop.

### PR DESCRIPTION
**Pull request by alchemixXx**
oop task - done

### CODE REVIEW

File: oop/homework.py [hunk 1]
```diff
@@ -39,31 +42,63 @@ class Cat:
...
 
     def __init__(self, age):
-        pass
+        self.age = age
+        self.average_speed = self._set_average_speed()
+        self.saturation_level = 50
 
     def eat(self, product):
-        pass
+        if product == 'fodder':
```
Comment (dimigor) [inline oop/homework.py]: what if U create dict with matching params `my_dict = {'apple': 5, .....}` 
and simply do that: `self_increase_satuation_level(my_dict.get(product, 0))`

File: oop/homework.py [hunk 2]
```diff
@@ -39,31 +42,63 @@ class Cat:
...
+        self.saturation_level += value
+        if self.saturation_level > 100:
+            self.saturation_level = 100
 
     def _set_average_speed(self):
-        pass
+        if self.age <= 7:
+            return 12
+        elif self.age <= 10:
+            return 9
```
Comment (dimigor) [inline oop/homework.py]: this func I think has to set property, no return
`this._some_property = 'new_value'`

File: oop/homework.py [hunk 3]
```diff
@@ -39,31 +42,63 @@ class Cat:
...
+            self._reduce_saturation_level(25)
+        else:
+            self._reduce_saturation_level(50)
+
+        return f"This ugly cat ran {distance} kilometers"
 
     def get_saturation_level(self):
-        pass
+        if self.saturation_level == 0:
+            return "Your cat is died :("
```
Comment (dimigor) [inline oop/homework.py:89]: maybe raise Exception?

File: oop/homework.py [hunk 4]
```diff
@@ -78,6 +113,22 @@ class Cheetah:
 
     """
 
+    def eat(self, product):
+        if product == 'gazelle':
```
Comment (dimigor) [inline oop/homework.py]: as above

File: oop/homework.py [hunk 5]
```diff
@@ -78,6 +113,22 @@ class Cheetah:
...
+            self._increase_saturation_level(30)
+        elif product == 'rabbit':
+            self._increase_saturation_level(15)
+        else:
+            self._increase_saturation_level(0)
+
+    def _set_average_speed(self):
```
Comment (dimigor) [inline oop/homework.py:115]: as above

**Review by dimigor — CHANGES_REQUESTED** [review]

File: oop/homework.py (hunk 1 above)
Comment (alchemixXx) [reply oop/homework.py]: Great idea! Thanks!

File: oop/homework.py (hunk 2 above)
Comment (alchemixXx) [reply oop/homework.py]: done

File: oop/homework.py (hunk 3 above)
Comment (alchemixXx) [reply oop/homework.py:89]: skipped

File: oop/homework.py (hunk 4 above)
Comment (alchemixXx) [reply oop/homework.py]: done

File: oop/homework.py (hunk 5 above)
Comment (alchemixXx) [reply oop/homework.py:115]: done

**Review by dimigor — APPROVED** [review]

**Pull request merged**